src/scales/scale.linear.js

```diff
@@ -70,20 +70,25 @@ module.exports = LinearScaleBase.extend({
 
 				if (chart.isDatasetVisible(datasetIndex) && IDMatches(meta)) {
 					helpers.each(dataset.data, function(rawValue, index) {
-						var value = +me.getRightValue(rawValue);
-						if (isNaN(value) || meta.data[index].hidden) {
+						var value = me._parseValue(rawValue);
+
+						if (isNaN(value.min) || isNaN(value.max) || meta.data[index].hidden) {
 							return;
 						}
 
 						positiveValues[index] = positiveValues[index] || 0;
 						negativeValues[index] = negativeValues[index] || 0;
 
+						if (value.min === 0 && !opts.ticks.beginAtZero) {
+							value.min = value.max;
+						}
+
 						if (opts.relativePoints) {
 							positiveValues[index] = 100;
-						} else if (value < 0) {
-							negativeValues[index] += value;
+						} else if (value.min < 0 || value.max < 0) {
```

Comment:
Would users ever want a bar that spans 0 to be in the positive stack? Do we need an additional config option here?

Reply:
I think I'd lean towards keeping the cases simpler for now and waiting until someone requests the functionality before adding additional config options